testing/file_test/testdata/alternating_files.carbon

```diff
@@ -0,0 +1,24 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+// AUTOUPDATE
+// CHECK:STDOUT: 3 args: `default_args`, `a.carbon`, `b.carbon`
+// CHECK:STDOUT: unattached message 1
+// CHECK:STDERR: unattached message 1
+
+// --- a.carbon
+// CHECK:STDERR: a.carbon:[[@LINE+1]]: message 2
+aaa
+// CHECK:STDOUT: a.carbon:[[@LINE-1]]: message 2
+
+// --- b.carbon
+// CHECK:STDERR: b.carbon:[[@LINE+4]]: message 3
+// CHECK:STDERR: a.carbon:2: message 4
+// CHECK:STDERR: b.carbon:[[@LINE+2]]: message 5
+// CHECK:STDERR: unattached message 6
+bbb
```

Comment:
```suggestion
bbb
bbbbbb
```

Sorry, I was thinking about this looking at the multifile testing, and wanted to suggest the edit as a really small nuance that should change how stdout is printed in what I think is a significant way.

It might be good too if you print something like `// CHECK:STDOUT: a.carbon:2: message 5.5`, since that presumably gets placed differently than the line sandwiched between two @LINE lines.

Reply:
I mean, this behavior is consistent with what we'd want for diagnostics, so maybe it's right. But I do think it's worth being explicit on it. :)

Reply:
I've made the documentation explicit about this.